src/main/java/seedu/address/model/travelplan/UniqueTravelPlanList.java

```diff
@@ -96,7 +99,7 @@ public void setTravelPlans(List<TravelPlan> travelPlans) {
             throw new DuplicateTravelPlanException();
         }
 
-        internalList.setAll(travelPlans);
+        internalList.setAll(travelPlans.stream().map(tp -> new TravelPlan(tp)).collect(Collectors.toList()));
```

Comment:
This is to ensure the internalList does not hold the exact same copy of the TravelPlan.